### PR TITLE
Finish the skeleton of the api for nomos-node

### PR DESCRIPTION
**Pull request by al8n**
May be we can change the crate name to `nomos-api`?  Maybe we also support `grpc` endpoint in the future?

The core logic of the PR is not large, but there is a large test case, to test if the swagger docs route works.

### CODE REVIEW

File: nomos-http-api/tests/todo.rs [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+        }
+    }
+}
+
+pub struct TodoEndpoint {
+    addr: SocketAddr,
+}
+
+#[async_trait::async_trait]
+impl Endpoint for TodoEndpoint {
```
Comment (danielSanchezQ) [inline nomos-http-api/tests/todo.rs:62]: This would be more a `Webserver` than an endpoint right?

Comment (danielSanchezQ) [reply nomos-http-api/tests/todo.rs:62]: In fact we could call it `Backend` so it is in the same fashion as other services (?)

Comment (al8n) [reply nomos-http-api/tests/todo.rs:62]: Yeah, it is a web server.

**Review by danielSanchezQ — APPROVED** [review]
Nicely contained!

File: nomos-http-api/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#[derive(Debug, Clone)]
+pub struct ApiServiceSettings<S> {
+    pub server_settings: S,
+}
+
+pub struct ApiService<E: Backend> {
+    settings: ApiServiceSettings<E::Settings>,
+}
+
+impl<E: Backend> ServiceData for ApiService<E> {
```
Comment (youngjoon-lee) [inline nomos-http-api/src/lib.rs:34]:
```suggestion
pub struct ApiServiceSettings<S> {
    pub backend_settings: S,
}

pub struct ApiService<B: Backend> {
    settings: ApiServiceSettings<B::Settings>,
}

impl<B: Backend> ServiceData for ApiService<B> {
```
we could probably rename like this?

Comment (youngjoon-lee) [reply nomos-http-api/src/lib.rs:34]: And, `endpoint` -> `backend` at line 59.

**Review by youngjoon-lee — APPROVED** [review]
Looks nice!

File: nomos-http-api/src/lib.rs (hunk 2 above)
Comment (danielSanchezQ) [reply nomos-http-api/src/lib.rs:34]: Nice catch!

Comment (al8n) [reply nomos-http-api/src/lib.rs]: Thank you! I forgot to change the generic param.

**Pull request merged**